tests/test_cases/test_cocotb/test_handle.py

```diff
@@ -66,21 +66,172 @@ async def test_delayed_assignment_still_errors(dut):
+    await int_values_test(dut.stream_in_data_39bit, [value])
+
+
+@cocotb.test()
+async def test_int_64bit(dut):
+    """Test int access to 64-bit vector."""
+    values = gen_int_test_values(len(dut.stream_in_data_wide))
+    await int_values_test(dut.stream_in_data_wide, values)
+
+
+@cocotb.test(expect_error=OverflowError)
+async def test_int_64bit_overflow(dut):
+    """Test 64-bit vector overflow."""
+    value = gen_int_ovfl_value(len(dut.stream_in_data_wide))
+    await int_values_test(dut.stream_in_data_wide, [value])
+
+
+@cocotb.test(expect_error=OverflowError)
+async def test_int_64bit_underflow(dut):
+    """Test 64-bit vector underflow."""
+    value = gen_int_unfl_value(len(dut.stream_in_data_wide))
+    await int_values_test(dut.stream_in_data_wide, [value])
+
+
+@cocotb.test()
+async def test_int_128bit(dut):
+    """Test int access to 128-bit vector."""
+    values = gen_int_test_values(len(dut.stream_in_data_dqword))
+    await int_values_test(dut.stream_in_data_dqword, values)
+
+
+@cocotb.test(expect_error=OverflowError)
+async def test_int_128bit_overflow(dut):
+    """Test 128-bit vector overflow."""
+    value = gen_int_ovfl_value(len(dut.stream_in_data_dqword))
+    await int_values_test(dut.stream_in_data_dqword, [value])
+
+
+@cocotb.test(expect_error=OverflowError)
+async def test_int_128bit_underflow(dut):
+    """Test 128-bit vector underflow."""
+    value = gen_int_unfl_value(len(dut.stream_in_data_dqword))
+    await int_values_test(dut.stream_in_data_dqword, [value])
+
+
 @cocotb.test(expect_error=AttributeError if cocotb.SIM_NAME in ["Icarus Verilog"] else ())
 async def test_integer(dut):
-    """
-    Test access to integers
-    """
-    log = logging.getLogger("cocotb.test")
-    await Timer(10, "ns")
-    dut.stream_in_int = 4
-    await Timer(10, "ns")
-    await Timer(10, "ns")
-    got_in = int(dut.stream_out_int)
-    got_out = int(dut.stream_in_int)
-    log.info("dut.stream_out_int = %d" % got_out)
-    log.info("dut.stream_in_int = %d" % got_in)
-    assert got_in == got_out, "stream_in_int and stream_out_int should not match"
+    """Test access to integers."""
+    for value in [0, 1, -1, 4, -4, 2**31-1, -2**31]:
+        dut.stream_in_int = value
+        await cocotb.triggers.Timer(10, 'ns')
+        assert int(dut.stream_in_int) == value
+
+
+@cocotb.test(expect_error=AttributeError if cocotb.SIM_NAME in ["Icarus Verilog"] else OverflowError)
+async def test_integer_overflow(dut):
+    """Test integer overflow."""
+    value = 2**31
+    dut.stream_in_int = value
+    await cocotb.triggers.Timer(10, 'ns')
+    assert int(dut.stream_in_int) == value
+
+
+@cocotb.test(expect_error=AttributeError if cocotb.SIM_NAME in ["Icarus Verilog"] else OverflowError)
+async def test_integer_underflow(dut):
+    """Test integer underflow."""
+    value = -2**32-1
+    dut.stream_in_int = value
+    await cocotb.triggers.Timer(10, 'ns')
+    assert int(dut.stream_in_int) == value
```

Comment:
`OverflowError` should be caught in the test, as:
```suggestion
@cocotb.test(expect_error=AttributeError if cocotb.SIM_NAME in ["Icarus Verilog"] else ())
async def test_integer_underflow(dut):
    """Test integer underflow."""
    value = -2**32-1
    with assert_raises(OverflowError):
        dut.stream_in_int = value
```

Same comment applies to all these tests.

Reply:
OK I see, I'll fix this up